src/ldp/representation/RepresentationMetadata.ts

```diff
@@ -1,47 +1,219 @@
+    return this;
+  }
+
   /**
-   * Optional content type of the representation.
+   * Finds all object values matching the given predicate.
+   * @param predicate - Predicate to get the values for.
+   *
+   * @returns An array with all matches.
    */
-  contentType?: string;
+  public getAll(predicate: NamedNode | string): Term[] {
+    return this.store.getQuads(this.id, this.convertPredicate(predicate), null, null)
+      .map((quad): Term => quad.object);
+  }
+
   /**
-   * Optional encoding of the representation.
+   * @param predicate - Predicate to get the value for.
+   *
+   * @throws Error
+   * If there are multiple matching values.
+   *
+   * @returns The corresponding value. Undefined if there is no match
    */
-  encoding?: string;
+  public get(predicate: NamedNode | string): Term | undefined {
+    const terms = this.getAll(predicate);
+    if (terms.length === 0) {
+      return;
+    }
+    if (terms.length > 1) {
+      throw new Error(`Multiple results for ${typeof predicate === 'string' ? predicate : predicate.value}`);
+    }
+    return terms[0];
+  }
+
   /**
-   * Optional language of the representation.
+   * Sets the value for the given predicate, removing all other instances.
+   * @param predicate - Predicate linking to the value.
+   * @param object - Value to set.
    */
-  language?: string;
+  public set(predicate: NamedNode | string, object: NamedNode | Literal | string): this {
+    this.removeAll(predicate);
+    this.add(predicate, object);
+    return this;
+  }
+
   /**
-   * Optional timestamp of the representation.
+   * Converts the incoming predicate to a named node.
+   * In case of string, first checks if it is a shortcut, if not a new named node gets made.
+   * @param predicate - Predicate to potentially transform.
    */
-  dateTime?: Date;
+  private convertPredicate(predicate: NamedNode | string): NamedNode {
+    if (typeof predicate === 'string') {
+      return shortcuts[predicate] || namedNode(predicate);
+    }
+    return predicate;
+  }
+
   /**
-   * Optional link relationships of the representation.
+   * Converts an object to a literal when needed.
+   * @param object - Object to potentially transform.
    */
-  linkRel?: { [id: string]: Set<string> };
+  private convertObject(object: NamedNode | Literal | string): NamedNode | Literal {
+    return typeof object === 'string' ? literal(object) : object;
+  }
+
+  // Syntactic sugar for common predicates
+
   /**
-   * Optional slug of the representation.
-   * Used to suggest the URI for the resource created.
+   * Shortcut for the CONTENT_TYPE predicate.
    */
-  slug?: string;
+  public get contentType(): string | undefined {
+    return this.get(shortcuts.contentType)?.value;
+  }
+
+  public set contentType(input) {
+    this.removeAll(shortcuts.contentType);
+    if (input) {
+      this.add(shortcuts.contentType, input);
+    }
+  }
```

Comment:
Isn't there a simple `this.set`?

Reply:
`.add` is slightly more efficient (since set calls remove and then add), but could change it for clarity.

Reply:
…but we're calling remove and add anyway? 🙂

Reply:
Ah my mistake, I was thinking of somewhere else in the code. This case is to allow the content type to be set to undefined again (which currently isn't supported with set since that would try to add the undefined value, could also change that).

Reply:
Yeah please do, since we want the code to these shortcut setters do be as easy as possible.